scipy/stats/_qmc.py

```diff
@@ -711,8 +711,8 @@ def __init__(
         optimization: Optional[Literal["random-cd", "lloyd"]] = None,
         seed: SeedType = None
     ) -> None:
-        if not np.issubdtype(type(d), np.integer):
-            raise ValueError('d must be an integer value')
+        if not np.issubdtype(type(d), np.integer) or d < 0:
```

Comment:
If `d=0` is to be considered ok, it should probably have tests if it g
Does not already. Also, technically this would be "non-negative" below.

Reply:
I rephrased, and we already have tests for the 0 dim case. It's done for all engines as we are checking the shape when doing random (0, n).